mixins/base-child.js

```diff
@@ -108,6 +108,10 @@ const baseChild = {
       .replace(/\/+/g, '/')
       .replace(/\/$/, '');
 
+    if (this.shopify.options.apiVersion) {
+      path = `/admin/api/${this.shopify.options.apiVersion}/${path.slice(7)}`;
```

Comment:
I would prefer something like this:

```js
let path = '/admin';

if (this.shopify.options.apiVersion) {
  path += `/api/${this.shopify.options.apiVersion}`;
}

path += `/${this.parentName}/${parentId}/${this.name}/${id}`;
path = path.replace(/\/+/g, '/').replace(/\/$/, '') + '.json';
```

We should also update resources that overrides `buildUrl()`. There are a few.

Reply:
Alright, that's slightly cleaner. I found overrides in shopify-payments and fulfillment-events, so I've updated those as well. I am working on adding tests, but I've never used instanbul/nyc or nock before, just dabbled a bit with mocha/chai, so it's slightly slow going.

Reply:
No problem, take you time.

Reply:
I'm not sure how I'd check to make sure the request path changed, since it looks like the `Shopify.request` method just returns the body. Is there a way to intercept it before it goes out?

Reply:
You can use nock. Use the updated path on it. See https://github.com/nock/nock#read-this---about-interceptors.

Reply:
Does it actually transform the request? I tried this, but it's hitting the `/admin/test` one, not the API versioned one. 

```
scope
  .get(`/admin/api/${apiVersion}/test`)
  .reply(200, { status: 'success' });

scope
  .get('/admin/test')
  .reply(400, { status: 'fail' });

const shopify = new Shopify({ shopName, accessToken, apiVersion });
const url = assign({ path: '/admin/test' }, shopify.baseUrl);

return shopify.request(url, 'GET', 'foo')
  .then((res) => {
    expect(res).to.equal({ status: 'success' });
  });
```

Reply:
you are not using the mixins `buildUrl()` in that example are you are directly calling `shopify.request()`. Try with a resource that actually uses the mixin and it should work.